src/classrooms/classrooms.service.ts

```diff
@@ -93,4 +93,17 @@ export class ClassroomsService {
     await this.em.removeAndFlush(classroom);
     return true;
   }
+
+  async updateClassroom(
+    id: number,
+    updateClassroomDto: UpdateClassroomDto,
+    userId: number,
+  ): Promise<Classroom | null> {
+    const classroom = await this.getClassroomById(id, userId);
+    if (!classroom) {
+      return null;
+    }
```

Comment:
Do we need this check? The above method should be calling findOneOrFail so it would throw an exception.

Additionally, I am not sure I understand why we are returning null when classroom does not exist.

Reply:
No bhaiya. I have refactor the code. I need to change things a bit